Mathlib/Data/MvPolynomial/Variables.lean

```diff
@@ -884,4 +890,18 @@ end EvalVars
 
 end CommSemiring
 
+section Field
+
+variable [Field F] {p q : MvPolynomial σ F}
```

Comment:
Can you generalize this to integral domains? Thanks!

Reply:
Ok I attempted this but had trouble, notes to self from the past hour or so of trying:

I see two possible routes. First route is to draw an equivalence to Polynomials over MvPolynomials in one less variable and apply the Polynomial version of this lemma. I ran into the issue that while `[MvPolynomial.optionEquivLeft](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/MvPolynomial/Equiv.html#MvPolynomial.optionEquivLeft)` exists, there was no obvious way to get the type into an option type. Luckily Floris came through with

```
import Mathlib.Logic.Equiv.Option

lemma Equiv.toOption (A : Type) (a : A) [DecidableEq A] : A ≃ Option {b : A // b ≠ a} :=
  Equiv.optionSubtype a |>.symm (.refl _) |>.1.symm
```

Unfortunately I now immediately run into the problem that, in this file, `optionEquivLeft` is a circular import. Also I would need to double check that the resulting MvPolynomial has the right typeclasses.

Second option would be to try to replicate the `Polynomial` version as closely as possible, with corresponding API. Problem there is that I immediately run into `Polynomial.degree_mul'` which references `leadingCoefficient`, which does not make sense for MvPolynomials. Not sure to what extent this can be routed around.

Happy to get any advice here.

Reply:
Ok looking into it more I see it's totally possible to split out the vars part of the file from the degrees part, so maybe that can come before all these other changes